### DISCUSSION TITLE
Help with STM32 drawing loop.

### DISCUSSION THREAD

**Discussion by Bridiro** [slint-ui/slint#7017] [Q&A] [answered]
Hi averyone, I've been stuck on this for a while and It's time to call for help. I'm on a STM32H747IIT6 and I'm trying to have the M7 to handle Slint and the screen, while my M4 will be reading data from CAN and various buttons. My problem is that I don't know how to put those data on the screen. I know How to get them on the M7, but once I start the drawing loop (from the template app_window->run() ), how do I constantly update them?

**Comment by ogoffart** [comment, marked as answer]
You got the Slint hello world running on the screen driven from the M7, right?

And you're wondering how to read the data from the M4 while the event loop is running?

That's a good question and we would need to add some example about that.

And the thing is that I even don't know the answer. I was about to say to use `slint::invoke_from_event_loop` from an interrupt, but i realize that this feature is not implemented for our STM32 platform backend.

So your best option right now is to add a timer with slint::Timer and poll from data from the memory in the timer handler.

**Reply by Bridiro** [reply]
My first tought was exactly to use `slint::invoke_from_event_loop`, but as I saw it was missing for STM32 I opted for slint::Timer. I left the discussion open to see if there was some better solutions but I think I'm going with this, thank you so much.
